lib/components/base-components/Renderable.ts

```diff
@@ -179,11 +179,27 @@ export abstract class Renderable implements IRenderable {
     // Skip if component is being removed and not initialized
     if (!isInitialized && this.shouldBeRemoved) return
 
+    const emitRenderEvent = (eventType: "start" | "end") => {
+      const eventPayload = { renderId: this._renderId }
+      const eventName = `renderable:renderLifecycle:${phase}:${eventType}`
+      ;(this as any).root
+        ?.emit(
+          eventName,
+          eventPayload,
+        )(this as any)
+        .root?.emit("renderable:renderLifecycle:anyEvent", {
+          ...eventPayload,
+          eventName,
```

Comment:
The Event standard specifies "type" as the key: https://developer.mozilla.org/en-US/docs/Web/API/Event

Can you also add `componentDisplayName: this.getString()` to help us identify the component being logged.

```suggestion
          type: eventName,
```